runtime/util/resolvehelp.c

```diff
@@ -78,9 +79,10 @@ getMethodForSpecialSend(J9VMThread *vmStruct, J9Class *currentClass, J9Class *re
 						method = *(J9Method **)(((UDATA)superclass) + vTableIndex);
 					}
 				} else {
-					/* 'methodClass' is a superclass of the current class so vTableIndex can't be 0 here */
-					Assert_Util_true(0 != vTableIndex);
-					method = *(J9Method **)(((UDATA)superclass) + vTableIndex);
+					/* In order to support non-vTable methods in a super invoke, perform the lookup rather than
+					 * looking in the vTable.
+					 */
+					method = (J9Method *)vmFuncs->javaLookupMethod(vmStruct, superclass, J9ROMMETHOD_NAMEANDSIGNATURE(J9_ROM_METHOD_FROM_RAM_METHOD(method)), currentClass, J9_LOOK_VIRTUAL | J9_LOOK_NO_VISIBILITY_CHECK | extraOptions);
```

Comment:
As this does a re-resolve in the superclass, is the code above which does a second vtable walk to find the most recent vtable slot required?

```
 				/* CMVC 170457: Algorithm for invokespecial lookup is wrong.
 				 * New Algorithm:
 				  * 1) Find the vtable index for J9Method in the resolved class.
 				 * 2) Using the vtable index to get the J9Method at that index in the current class.
 				 * 3) Walk the current class vtable backwards to find the most "recent" override of that method
 				 * 4) Lookup the method at vtableIndex in currentClass's super class
 				 */
 				method = *(J9Method **)(((UDATA)currentClass) + vTableIndex);	
 				vTableIndex = vmFuncs->getVTableIndexForMethod(method, currentClass, vmStruct);
```

Possibly for the `isInterfaceMethod` case...

Reply:
Could be written more optimally now, yes.

Reply:
Is that enthusiastic agreement to make it more optimal? =)